### ISSUE TITLE
HTTP API login with HTTP digest authentication

### ISSUE THREAD

**Issue by christianschneider89** [blakeblackshear/frigate#13297]
Please add the ability to log in to the API using HTTP digest authentication.

As asked in question https://github.com/blakeblackshear/frigate/discussions/13294, please add the ability to use HTTP digest authentication for logging into the HTTP API.

With the current implementation there is no possibility to use the API with just one single cURL request.
In some applications there is only one configurable option (e.g. for create an event: /api/events/[camera]/[eventname]/create)

**Comment by NickM-27** [comment]
If I had to guess it won't be done via digest but rather a long lived token, I will defer to Blake though

**Comment by nijave** [comment]
I use Contour/Envoy with ext_authz for this but any proxy will work https://docs.frigate.video/configuration/authentication/#proxy-configuration

Then the proxy can apply arbitrary auth rules

**Comment by Rudd-O** [comment]
The Prometheus metrics feature from Frigate is 100% useless if they can't be exposed unauthenticated or with some seriously long lived token, because Prometheus can't do authentication under the current circumstances.  Recommend deleting the useless feature altogether.

**Comment by nijave** [comment]
> The Prometheus metrics feature from Frigate is 100% useless if they can't be exposed unauthenticated or with some seriously long lived token, because Prometheus can't do authentication under the current circumstances. Recommend deleting the useless feature altogether.

As I said, it works fine with an auth proxy and a route exclusion. Definitely not ideal but the config is in the docs

**Comment by Rudd-O** [comment]
Yeah, well, that forces us to deploy not just a regular proxy but an auth proxy — with auth aligned to Frigate — in front of Frigate.
